File: Radar.client/src/models/Game.js


export class Game {
  constructor(data) {
    this.id = data.id;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    this.creator = data.creator;
    this.creatorId = data.creatorId;
    this.isPrivate = data.isPrivate;
    this.map = data.map;
    this.mapId = data.mapId;
    this.name = data.name;
    this.pin = data.pin;
    this.playerCount = data.playerCount;
    this.playerLimit = data.playerLimit;
    this.status = data.status;
    this.type = data.type;
    this.winner = data.winner;
    this.winnerId = data.winnerId;
  }
}

File: Radar/Models/Game.cs
namespace Radar.Models;

public class Game : IHasCreator
{
  // IHasCreator
  public string CreatorId { get; set; }
  public Account Creator { get; set; }

  // IDbItem
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public int Id { get; set; }

  // Game Specific
  public string Name { get; set; }
  public int MapId { get; set; }
  public int PlayerLimit { get; set; }
  public bool IsPrivate { get; set; }
  public int Pin { get; set; }
  public int PlayerCount { get; set; }
  public string Status { get; set; }
  public string Type { get; set; }
  public string WinnerId { get; set; }
  public Account Winner { get; set; }
  public Map Map { get; set; }
}